static/app/views/alerts/rules/metric/details/index.tsx

```diff
@@ -59,7 +59,8 @@ class MetricAlertDetails extends Component<Props, State> {
 
   componentDidUpdate(prevProps: Props) {
     if (
-      prevProps.location.search !== this.props.location.search ||
+      (prevProps.location.search !== this.props.location.search &&
+        prevProps.location.query.mute !== '1') ||
```

Comment:
Instead of specifically ignoring `mute` here, I think it might be better to compare the properties that should cause a refetch instead (alert, period, start, end). Something like:

```ts
const prevQuery = pick(prevProps.location.query, ['alert','period','start','end'])
const nextQuery = pick(props.location.query, ['alert','period','start','end'])
if (!isEqual(prevQuery, nextQuery) || prevProps.organization.slug !== this.props.organization.slug || ...) { ... }
```